backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -901,31 +901,44 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
             }
         }
 
-        // If object is imported - access it via getter function.
-        if (isExternal(irClass)) {
-            val valueGetterName = irClass.objectInstanceGetterSymbolName
-            val valueGetterFunction = LLVMGetNamedFunction(context.llvmModule, valueGetterName) ?:
-                LLVMAddFunction(context.llvmModule, valueGetterName,
-                        functionType(kObjHeaderPtr, false, kObjHeaderPtrPtr))
+        val storageKind = irClass.storageKind(context)
+
+        // If thread local object is imported - access it via getter function.
+        if (isExternal(irClass) && storageKind == ObjectStorageKind.THREAD_LOCAL) {
```

Comment:
It makes sense to expose a thread-local singleton by generating getter for the instance storage instead of getter for the instance itself. Would this make the compiler code simpler by unifying thread-local and global singletons?